gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala

```diff
@@ -174,7 +174,7 @@ trait GlutenTestsTrait extends GlutenTestsCommonTrait {
     }
     if (!(checkResult(result.head.get(0), expected, expression.dataType, expression.nullable)
       || checkResult(
-      CatalystTypeConverters.convertToCatalyst(result.head.get(0)),
+      CatalystTypeConverters.createToCatalystConverter(expression.dataType)(result.head.get(0)),
```

Comment:
Does this change indicate the result schema could be wrong? If so, we should add some notes here.